source/shared/cpp/ObjectModel/MarkDownParser.cpp

```diff
@@ -0,0 +1,478 @@
+#include <vector>
+#include <iomanip>
+#include <iostream>
+#include "MarkDownParser.h"
+
+using namespace AdaptiveCards;
+
+MarkDownParser::MarkDownParser(const std::string &txt) : m_text(txt), 
+        m_textBegin(m_text.begin()), m_curPos(m_text.begin()), m_textEnd(m_text.end())
+{
+}
+
+// adjust number of emphasis counts after maching is done
+int MarkDownParser::AdjustEmphasisCounts(int leftOver, std::list<Emphasis>::iterator left_itr, 
+        std::list<Emphasis>::iterator right_itr)
+{
+    int delimCnts = 0;
+    if (leftOver >= 0)
+    {
+        delimCnts = left_itr->m_emphCnts - leftOver;
+        left_itr->m_emphCnts = leftOver;
+        right_itr->m_emphCnts = 0;
+    }
+    else
+    {
+        delimCnts = left_itr->m_emphCnts;
+        right_itr->m_emphCnts = leftOver * (-1);
+        left_itr->m_emphCnts = 0;
+    }
+    return delimCnts;
+}
+
+// Following the rules speicified in CommonMark (http://spec.commonmark.org/0.27/)
+// It generally supports more stricker version of the rules
+// push left delims to stack, until matching right delim is found,
+// update emphasis counts and type to build string after search is complete
+std::string MarkDownParser::TransformToHtml()
+{
+    GenSymbolTable ();
```

Comment:
>  [](start = 18, length = 1)

[nit] there shouldn't be a space between the function name and the args, I see that in multiple places.